core/src/material.cpp

```diff
@@ -115,6 +111,23 @@ void Material::set_uniforms() {
 
     // clang-format off
 }
+void Material::set_texture_slot_path(TextureType slot, const std::string& path){
+switch (slot) {
```

Comment:
Would we add in the future some sort of TextureType that cannot be set? Cause if not, we could maybe replace this with a `static_assert` that checks it isn't the `LAST`. So all the errors would end up in compile time, maybe it won't work and it needs to be in runtime.